src/site_upload/process_flat/process_flat.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import os
+
+import awswrangler
+from shared import awswrangler_functions, decorators, enums, functions, pandas_functions, s3_manager
+
+log_level = os.environ.get("LAMBDA_LOG_LEVEL", "INFO")
+logger = logging.getLogger()
+logger.setLevel(log_level)
+
+
+def process_flat(manager: s3_manager.S3Manager):
+    new_file = False
+    if awswrangler.s3.does_object_exist(
+        f"s3://{manager.s3_bucket_name}/{manager.parquet_flat_key}"
+    ):
+        manager.move_file(
+            manager.parquet_flat_key,
+            manager.parquet_flat_key.replace(
+                enums.BucketPath.FLAT.value, enums.BucketPath.ARCHIVE.value
+            ),
+        )
+    else:
+        new_file = True
+    manager.move_file(
+        manager.s3_key,
+        manager.parquet_flat_key,
+    )
+    awswrangler_functions.generate_csv_from_parquet(
+        manager.s3_bucket_name,
+        manager.parquet_flat_key.split("/", 1)[0],
+        manager.parquet_flat_key.split("/", 1)[1],
+        f"s3://{manager.s3_bucket_name}/{manager.csv_flat_key}",
+    )
+    df = awswrangler.s3.read_parquet(f"s3://{manager.s3_bucket_name}/{manager.parquet_flat_key}")
+    column_dict = pandas_functions.get_column_datatypes(df.dtypes)
+    manager.update_local_metadata(
+        enums.ColumnTypesKeys.COLUMNS.value,
+        value=column_dict,
+        site=manager.site,
+        metadata=manager.types_metadata,
+        meta_type=enums.JsonFilename.COLUMN_TYPES.value,
+        extra_items={
+            "s3_path": f"s3://{manager.s3_bucket_name}/{manager.parquet_flat_key}",
+            "type": "flat",
+        },
+    )
+    manager.update_local_metadata(enums.TransactionKeys.LAST_DATA_UPDATE.value, site=manager.site)
+    manager.update_local_metadata(
+        enums.ColumnTypesKeys.LAST_DATA_UPDATE.value,
+        value=column_dict,
+        metadata=manager.types_metadata,
+        meta_type=enums.JsonFilename.COLUMN_TYPES.value,
+    )
+    manager.write_local_metadata(
+        metadata=manager.types_metadata, meta_type=enums.JsonFilename.COLUMN_TYPES.value
+    )
+    manager.write_local_metadata()
```

Comment:
Similar question here - the second call I assume ends up overwriting the first?

Reply:
See above - `meta_type` redirection.